exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroRecordReader.java

```diff
@@ -403,7 +403,9 @@ public void close() {
   public String toString() {
     long currentPosition = -1L;
     try {
-      currentPosition = reader.tell();
+      if (reader != null) {
+        currentPosition = reader.tell();
+      }
     } catch (IOException e) {
       logger.trace("Unable to obtain reader position: " + e.getMessage());
```

Comment:
Please log the stack trace of the exception.

Reply:
Done.